admin/preprint_providers/forms.py

```diff
@@ -0,0 +1,32 @@
+from django.forms import ModelForm, CheckboxSelectMultiple, MultipleChoiceField, HiddenInput, CharField
+
+from osf.models import PreprintProvider, Subject
+from admin.base.utils import get_subject_rules, get_toplevel_subjects, get_nodelicense_choices
+
+
+class PreprintProviderForm(ModelForm):
+    toplevel_subjects = MultipleChoiceField(widget=CheckboxSelectMultiple())
+    subjects_chosen = CharField(widget=HiddenInput())
+
+    class Meta:
+        model = PreprintProvider
+
+        exclude = ['modm_model_path', 'modm_query', '_id']
+
+        widgets = {
+            'licenses_acceptable': CheckboxSelectMultiple(),
+            'subjects_acceptable': HiddenInput(),
+        }
+
+    def __init__(self, *args, **kwargs):
+        toplevel_choices = get_toplevel_subjects()
+        nodelicense_choices = get_nodelicense_choices()
+        super(PreprintProviderForm, self).__init__(*args, **kwargs)
+        self.fields['toplevel_subjects'].choices = toplevel_choices
```

Comment:
The tests already should have db access. You shouldn't call the function because that would result in database access upon *importing* the file.

Have you tried passing choices to `MultipleChoiceField`? According to the [docs](https://docs.djangoproject.com/en/1.10/ref/forms/fields/#django.forms.ChoiceField.choices), this is allowed.